src/drivers/localstorage.js

```diff
@@ -123,7 +124,7 @@
                     value = serializer.deserialize(value);
                 }
 
-                value = iterator(value, key.substring(keyPrefixLength), i + 1);
+                value = iterator(value, key.substring(keyPrefixLength), iterationNumber++);
```

Comment:
Thank you for your efforts on this matter.

Indeed, the problem was that we should skip the non-localForage items from
the iterationNumber count.

Also, I have noticed the same as Thodoris about the order of items that the
`iterate` brings, it seems to be the opposite order (it does not affect the
iterationNumber count though), so the test is OK.

On Thu, Aug 27, 2015 at 2:33 AM, Matthew Riley MacPherson <
notifications@github.com> wrote:

> In src/drivers/localstorage.js
> <https://github.com/mozilla/localForage/pull/435#discussion_r38077853>:
>
> > @@ -123,7 +124,7 @@
> >                      value = serializer.deserialize(value);
> >                  }
> >
> > -                value = iterator(value, key.substring(keyPrefixLength), i + 1);
> > +                value = iterator(value, key.substring(keyPrefixLength), iterationNumber++);
>
> Weird! Interesting. Okay, I'll fix that up and try to do it now. I've
> gotta pack for holiday though, so if I don't do it now I'll get to it
> tonight [image: :smile:]
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mozilla/localForage/pull/435/files#r38077853>.
>